### PR TITLE
Organize executable metadata code

### PR DESCRIPTION
**Pull request by RobertBColton**
A couple of changes here to address #1156 so that the Portable ZIP continues working.

1. Wrapped the chmod call for the desktop entry in a preprocessor guard. Even if you are cross-compiling from Windows for Linux, @fundies says there's no reason to call this function because the filesystem is NTFS and it makes no sense. However, newer MinGW can compile this, so they probably have some sort of compatibility logic. Not sure if this is the correct thing we want to do?
2. Moved the Windows exe metadata writing into a helper function.
3. Check the actual target before writing either the Windows exe metadata or the desktop entry. Windows does not check the mode because the icon works in run and debug mode.
4. Changed tabs to 2 spaces in my Windows exe metadata writing code only.
5. Readded `output_log.txt` to `.gitignore`

### CODE REVIEW

File: CompilerSource/compiler/compile.cpp [hunk 1]
```diff
@@ -112,8 +112,42 @@ inline void write_desktop_entry(const std::string fPath, const GameSettings& gam
   wto << "Categories=Game;\n";
   wto << "MimeType=application/octet-stream;\n";
   wto.close();
-  
+
+  #if defined(__linux__)
```
Comment (fundies) [inline CompilerSource/compiler/compile.cpp]: i told you ifndef win32 idiot. windows is the only os without chmod. osx has it

Comment (RobertBColton) [reply CompilerSource/compiler/compile.cpp]: Does it have the same concept of executable files though?

File: CompilerSource/compiler/compile.cpp [hunk 2]
```diff
@@ -118,7 +118,8 @@ inline void write_desktop_entry(const std::string fPath, const GameSettings& gam
   #endif
 }
 
-inline void write_exe_info(const std::string makedir, const EnigmaStruct *es) {
+inline void write_exe_info(const std::string makedir, const EnigmaStruct *es)
+{
```
[the remark this answers is not in the document]
Comment (RobertBColton) [reply CompilerSource/compiler/compile.cpp]: Sure, but the rest of the file has this style, so I either make it conform or reformat the whole file? Which would you prefer captain fundies?

**Comment by RobertBColton** [comment]
@fundies Any particular reason you chose to not write the desktop entry in run or debug mode? It should actually work and set the correct icon if you did, all the icon is a string to the file path. However, I think Linux uses PNG not necessarily ICO, and I haven't exactly added the UI options to LGM yet.

File: CompilerSource/compiler/compile.cpp (hunk 1 above)
Comment (fundies) [reply CompilerSource/compiler/compile.cpp]: yes, in all fillesystems other than ntfs ad fat32 theres an extecutable bit on files. Windows just dumb

File: CompilerSource/compiler/compile.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (fundies) [reply CompilerSource/compiler/compile.cpp]: this whole file needs rewritting. I think it's the ugliest file in our tree

**Comment by fundies** [comment]
Because you dont need the desktop entry to run the game. It's really only useful to distribute with a release

**Comment by RobertBColton** [comment]
Sure but if it does attach the game icon correctly once we add the UI setting, I'd rather it just write it out that way then jumping through hoops to set icon. It is honestly useful to have the game icon in run/debug if you have a lot of things open in the taskbar/launcher.

**Comment by fundies** [comment]
this has nothing todo with the favicon

**Comment by RobertBColton** [comment]
I assume by favicon you mean the Window's frame icon, on the frame of the window with the close/min/max buttons? If that is what you meant, then I didn't know Linux had that. As I recall, Ubuntu's DE or w/e only showed the icon in the launcher. So in this case I thought favicon was synonymous with game icon and window icon etc.

File: CompilerSource/compiler/compile.cpp (hunk 1 above)
Comment (RobertBColton) [reply CompilerSource/compiler/compile.cpp]: Is it at all possible on Linux to download a file already chmod'd to be executable? What exactly does chmod do? Does it change anything in the actual bytes that comprise the file or is it changing something in the operating system's files/the file system's table?

Comment (RobertBColton) [reply CompilerSource/compiler/compile.cpp]: Asked Josh, and I have changed it: 2595c1da18a2843b80faa595263dd8eb562f4d5b

File: CompilerSource/compiler/compile.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (RobertBColton) [reply CompilerSource/compiler/compile.cpp]: Josh said just use correct style on new codes only, fixed: 2595c1da18a2843b80faa595263dd8eb562f4d5b

**Comment by fundies** [comment]
Don't wanna merge cause itt'll conflick with my latest pr and I cba to fix

**Comment by RobertBColton** [comment]
Whatever, I have no problem with fixing merge conflicts, it's easy.

**Comment by fundies** [comment]
you can resolve conflict now :P

**Pull request merged**